src/java.base/share/classes/java/io/BufferedInputStream.java

```diff
@@ -643,9 +643,13 @@ private long implTransferTo(OutputStream out) throws IOException {
         if (getClass() == BufferedInputStream.class && markpos == -1) {
             int avail = count - pos;
             if (avail > 0) {
-                // Prevent poisoning and leaking of buf
-                byte[] buffer = Arrays.copyOfRange(getBufIfOpen(), pos, count);
-                out.write(buffer);
+                if (isTrusted(out)) {
```

Comment:
It might be cleaner for now to drop `isTrusted()` and just do the class check explicitly here. That still takes care of the main intent of not passing the buffer to an untrustworthy stream. Further cleanup and consolidation can be done later.

Reply:
I think this would be doing double job, wouldn't it?

Reply:
> I think this would be doing double job, wouldn't it?

Sorry, I don't understand.

Reply:
If we drop the method for now I have to write it later again, I guess

Reply:
> If we drop the method for now I have to write it later again, I guess

Maybe, but it's archived so it's easy enough.

Reply:
If this is only used after Module system is ready, we can use something like:
https://github.com/openjdk/jdk/blob/34351b7a7950a3b563748f40f2619374f62f9b16/src/java.base/share/classes/java/nio/file/Files.java#L3352

```java
if (out.getClass().getModule() == Object.class.getModule())
```

Reply:
I propose we simply go with the original solution, so we stay sync with BAIS.